### PR TITLE
Priority queue supports deprecated compute budget ix 

### LINKED ISSUE #32549: Using Compute Budget ix `requestUnits` fails on 1.16
#### Problem

Using the `requestUnits` ix on the Compute Budget program times out when sending transaction via RPC on Devnet.

Cluster: Devnet

Reproducible code:
```javascript
import {
  Keypair,
  Connection,
  LAMPORTS_PER_SOL,
  sendAndConfirmTransaction,
  ComputeBudgetProgram,
  SystemProgram,
  Transaction,
  clusterApiUrl,
} from "@solana/web3.js";

(async () => {
  const payer = Keypair.generate();
  const toAccount = Keypair.generate().publicKey;

  const connection = new Connection(clusterApiUrl("devnet"), "confirmed");

  const airdropSignature = await connection.requestAirdrop(
    payer.publicKey,
    LAMPORTS_PER_SOL
  );

  await connection.confirmTransaction(airdropSignature);

  const modifyComputeUnits = ComputeBudgetProgram.requestUnits({
    units: 1_000_000,
    additionalFee: 0,
  });

  const transaction = new Transaction()
    .add(modifyComputeUnits)
    .add(
      SystemProgram.transfer({
        fromPubkey: payer.publicKey,
        toPubkey: toAccount,
        lamports: 10000,
      })
    );

  const signature = await sendAndConfirmTransaction(connection, transaction, [
    payer,
  ]);
  console.log(signature);
  const result = await connection.getParsedTransaction(signature);
  console.log(result);
})();
```

IXs `setComputeUnitLimit` and `setComputeUnitPrice` still work fine. 

#### Proposed Solution

The ix `requestUnits` should not break until feature `EfhYd3SafzGT472tYQDUc4dPd2xdEfKs5fwkowUgVt4W` is activated.

### PR DESCRIPTION
**Pull request by taozhu-chicago**
#### Problem
Transaction without priority details will be dropped by priority queue (at `immutable_deserialized_packet::new()`), yet transaction_priority_details is hardcoded not to support deprecated ix. Combined, it results TX with deprecated compute budget ix will be dropped by leader. 

#### Summary of Changes
1. Add test case for transaction with deprecated ix;
2. change implementation to support above two scenarios.

Fixes #32549

### CODE REVIEW

**Comment by taozhu-chicago** [comment]
@willhickey

File: runtime/src/transaction_priority_details.rs [hunk 1]
```diff
@@ -28,12 +28,12 @@ pub trait GetTransactionPriorityDetails {
...
                 instructions,
                 true,  // use default units per instruction
-                false, // stop supporting prioritization by request_units_deprecated instruction
+                true,  // supports prioritization by request_units_deprecated instruction
                 true,  // enable request heap frame instruction
                 true,  // enable support set accounts data size instruction
                        // TODO: round_compute_unit_price_enabled: bool
             )
-            .ok()?;
+            .unwrap_or_default();
```
Comment (t-nelson) [inline runtime/src/transaction_priority_details.rs:36]: seems like flipping the hardcoded feature gate is the minimal fix?  we should do anything more than that in a follow up to reduce backport exposure

**Review by t-nelson — COMMENTED** [review]
figured as much :wink:

File: runtime/src/transaction_priority_details.rs (hunk 1 above)
Comment (taozhu-chicago) [reply runtime/src/transaction_priority_details.rs:36]: It actually makes better sense to keep `ok()?` -- if tx fails on processing compute budget ixs, it'd never be executed anyway, may as well just drop upfront. undoing this line of change...

**Review by t-nelson — COMMENTED** [review]
r+ @apfitzge for final approval

File: runtime/src/transaction_priority_details.rs (hunk 1 above)
Comment (t-nelson) [reply runtime/src/transaction_priority_details.rs]: makes sense, thanks!  `.unwrap_or_default()` has not been friendly in the past :wink:

File: runtime/src/transaction_priority_details.rs [hunk 2]
```diff
@@ -27,11 +27,11 @@ pub trait GetTransactionPriorityDetails {
         let prioritization_fee_details = compute_budget
             .process_instructions(
                 instructions,
-                true,  // use default units per instruction
-                false, // stop supporting prioritization by request_units_deprecated instruction
-                true,  // enable request heap frame instruction
-                true,  // enable support set accounts data size instruction
-                       // TODO: round_compute_unit_price_enabled: bool
+                true, // use default units per instruction
+                true, // supports prioritization by request_units_deprecated instruction
```
Comment (apfitzge) [inline runtime/src/transaction_priority_details.rs:31]: Kinda worried about putting true here.
Before we remove the code but after the feature is activated this means all txs using this unsupported ix get boosted to the front of our queue w/o hope of ever getting executed or paying fees (they will later be dropped when we try to execute).

Comment (taozhu-chicago) [reply runtime/src/transaction_priority_details.rs:31]: To clarify, after feature gate is enabled, and before code is removed. A TX with deprecated ix would: 
1. be placed into priority queue based on priority specified by deprecated ix
2. then being dropped by leader threads during packing (which is before loading and execution).

Later after feature code is removed, the same transaction will not be inserted into priority queue. 

So there will be some queuing inefficiency during the window of activated feature code removal.

Comment (apfitzge) [reply runtime/src/transaction_priority_details.rs:31]: > then being dropped by leader threads during packing (which is before loading and execution).

I think it'd be dropped during execution - after locks are taken and as first step in execution.

**Review by apfitzge — APPROVED** [review]
lgtm - this should be backported to v1.16.

Looks like this was introduced originally in #27503 (v1.15 - now abandoned) so no need to backport to v1.14 afaict.

**Pull request merged**